components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantRegistryDataDeletionUtil.java

```diff
@@ -26,94 +26,105 @@
 public class TenantRegistryDataDeletionUtil {
+
     public static final Log log = LogFactory.getLog(TenantRegistryDataDeletionUtil.class);
-    
+
+    private static final String DELETE_CLUSTER_LOCK = "DELETE FROM REG_CLUSTER_LOCK WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_LOG = "DELETE FROM REG_LOG WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_ASSOCIATION = "DELETE FROM REG_ASSOCIATION WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_SNAPSHOT = "DELETE FROM REG_SNAPSHOT WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_RESOURCE_COMMENT = "DELETE FROM REG_RESOURCE_COMMENT WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_COMMENT = "DELETE FROM REG_COMMENT WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_RESOURCE_RATING = "DELETE FROM REG_RESOURCE_RATING WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_RATING = "DELETE FROM REG_RATING WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_RESOURCE_TAG = "DELETE FROM REG_RESOURCE_TAG WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_TAG = "DELETE FROM REG_TAG WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_RESOURCE_PROPERTY = "DELETE FROM REG_RESOURCE_PROPERTY WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_PROPERTY = "DELETE FROM REG_PROPERTY WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_RESOURCE_HISTORY = "DELETE FROM REG_RESOURCE_HISTORY WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_CONTENT_HISTORY = "DELETE FROM REG_CONTENT_HISTORY WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_RESOURCE = "DELETE FROM REG_RESOURCE WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_CONTENT = "DELETE FROM REG_CONTENT WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_PATH = "DELETE FROM REG_PATH WHERE REG_TENANT_ID = ?";
+    private static final String DELETE_ADMIN_REGISTRY_RESORUCE = "DELETE FROM REG_RESOURCE WHERE REG_PATH_ID IN " +
+            "(SELECT DISTINCT REG_PATH_ID FROM REG_PATH WHERE REG_PATH_VALUE LIKE ?)";
+    private static final String DELETE_ADMIN_REGISTRY_PATH = "DELETE FROM REG_PATH WHERE REG_PATH_VALUE LIKE ?";
+
     /**
-     * Delete all tenant information related to tenant stored in REG tables
+     * Delete all tenant information related to tenant stored in REG tables.
+     *
      * @param tenantId id of tenant whose data should be deleted
-     * @param conn database connection object
-     * @throws SQLException thrown if an error occurs while executing the queries 
+     * @param conn     database connection object
+     * @throws SQLException thrown if an error occurs while executing the queries
      */
-    public static void deleteTenantRegistryData(int tenantId, Connection conn) throws Exception {
+    protected static void deleteTenantRegistryData(int tenantId, Connection conn) throws SQLException {
```

Comment:
the public method has made protected. If custom components has used this method, they will break